mcode/pims
# Add RxFill dispense support to PIMS

## Describe your changes

> Please include a summary of the changes and the related issue/task. Please also include relevant motivation and context. List any dependencies that are required for this change, including links to other pull requests/branches in other repositories if applicable.

* Create new mongoose model for NCPDP NewRx and POST to EHR from updateRx/:id/pickedUp using existing model's DoctorOrder.PrescriberOrderNumber
* Update buildRxStatus with fields from the NewRx model's serialized JSON instead of the DoctorOrder model and remove hardcoded values
* Remove comments/unused code related to DoctorOrder and fix console.error
* Remove unused file

## Issue ticket number and Jira link

> Please include the Jira Ticket Number and Link for this issue/task.

https://jira.mitre.org/browse/REMS-595

## How to manually test
1. On test-ehr - `git checkout rxfill` on commit b98825
2. On request-generator - `git checkout main` on commit efd4e7
3. Go to http://localhost:3000/, click launch, verify with alice/alice
4. Request a new medication for Jon Snow (Turalio) and go to the Network tab in DevTools
5. Check that the Payload for addRx contains `<To Qualifier="P">Pharmacy 123</To>`
6. Sign the order and fill out the Patient Enrollment Form, Prescriber Enrollment Form, and Prescriber Knowledge Assessment Form
7. Go to http://localhost:5050/DoctorOrders and go to the Network tab in DevTools
8. Click "Verify Order" on the New Orders tab, go to the Verified Orders tab, and click "Mark as Picked Up"
9. Check that in the terminal running the pims backend, `Sent RxFill status to EHR` appears with the test-ehr endpoint and the response.config.data (RxFill) and response.data (status) contain XML. 
10. Check that in the terminal running test-ehr, test-ehr has received the RxFill

## Checklist before requesting a review

## Checklist for conducting a review

## Workflow 

Owner of the Pull Request will be responsible for merge after all requirements are met, including approval from at least one reviewer. Additional changes made after a review will dismiss any approvals and require re-review of the additional updates. Auto merging can be enabled below if additional changes are likely not to be needed. The bot will auto assign reviewers to your Pull Request for you.
